### PR TITLE
{172983130}: Tweaking oldfile list

### PR DESCRIPTION
**Pull request by riverszhang89**
A few tweaks to the oldfile list logic to make it collect faster.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>7/532 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>schemalk_extralock_generated<br>schemalk<br>truncatesc_offline_generated<br>comdb2sys<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated</p>

File: bdb/llmeta.c [hunk 1]
```diff
@@ -10124,6 +10124,49 @@ static int bdb_process_each_entry(bdb_state_type *bdb_state, tran_type *tran, vo
...
+    uint8_t nxt[LLMETA_IXLEN];
+    int rc;
+    int irc = 0;
+    void *searchkey;
+    int searchkeylen;
+
+    assert(klen <= *rlen);
+    assert(memcmp(key, *resume, klen) == 0);
+
+    if (*resume == NULL) {
```
Comment (chands10) [inline bdb/llmeta.c:10143]: if *resume is NULL would this be valid for memcp in the line above?

**Review by chands10 — APPROVED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>10/534 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_resume_logicalsc_generated<br>schemalk_extralock_generated<br>phys_rep_tiered_nosource_generated<br>phys_rep_tiered<br>sc_lotsoftables<br>sc_lotsoftables_logicalsc_generated<br>truncatesc_offline_generated<br>comdb2sys<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated</p>

**Comment by riverszhang89** [comment]
@chands10 Thanks for the review!

**Pull request merged**